### PR TITLE
Add 4.12 z-rollback test for nightly builds

### PR DESCRIPTION
**Pull request by shellyyang1989**
Refers [OTA-1084](https://issues.redhat.com/browse/OTA-1084)

### CODE REVIEW

**Comment by shellyyang1989** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-aws-ipi-shared-vpc-phz-upgrade-rollback-oldest-supported

**Comment by shellyyang1989** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-aws-ipi-shared-vpc-phz-upgrade-rollback-oldest-supported

**Comment by shellyyang1989** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-azure-upi-f28-upgrade-rollback-oldest-supported

**Comment by shellyyang1989** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-gcp-ipi-ovn-ipsec-upgrade-rollback-oldest-supported

**Comment by shellyyang1989** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-ibmcloud-ipi-upgrade-rollback-oldest-supported

**Comment by shellyyang1989** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-vsphere-upi-upgrade-rollback-oldest-supported

**Comment by shellyyang1989** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-nutanix-ipi-upgrade-rollback-oldest-supported

**Comment by shellyyang1989** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-gcp-ipi-ovn-ipsec-upgrade-rollback-oldest-supported

**Comment by shellyyang1989** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-aws-ipi-byo-route53-upgrade-rollback-oldest-supported

**Comment by shellyyang1989** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-azure-upi-f28-upgrade-rollback-oldest-supported

**Comment by shellyyang1989** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-gcp-ipi-ovn-ipsec-upgrade-rollback-oldest-supported

**Comment by shellyyang1989** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-ibmcloud-ipi-upgrade-rollback-oldest-supported

**Comment by shellyyang1989** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-nutanix-ipi-upgrade-rollback-oldest-supported

**Comment by shellyyang1989** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-vsphere-upi-upgrade-rollback-oldest-supported

**Comment by shellyyang1989** [comment]
All the rehearsals failed. I checked the [aws](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/46465/rehearse-46465-periodic-ci-openshift-release-master-nightly-4.12-aws-ipi-byo-route53-upgrade-rollback-oldest-supported/1732948738046955520) and [gcp](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/46465/rehearse-46465-periodic-ci-openshift-release-master-nightly-4.12-gcp-ipi-ovn-ipsec-upgrade-rollback-oldest-supported/1732948871534874624) runs, both failed with the error.
```
Dec 08 05:44:35.712853 ip-10-0-58-39 rpm-ostree[5775]: Failed to find image ostree-unverified-image:docker://quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:517eb21c6ba447a4e203b60578f3af1c11a1066863831d02d48dd1d34752f82b  
Dec 08 05:44:35.715874 ip-10-0-58-39 rpm-ostree[5775]: error: Couldn't start daemon: Error setting up sysroot: Reading deployment 1: Failed to find image ostree-unverified-image:docker://quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:517eb21c6ba447a4e203b60578f3af1c11a1066863831d02d48dd1d34752f82b
```

**Comment by shellyyang1989** [comment]
@LalatenduMohanty @wking @petr-muller  @jianlinliu @jiajliu @JianLi-RH Please help review when you have time. Thank you.

**Comment by petr-muller** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-gcp-ipi-ovn-ipsec-upgrade-rollback-oldest-supported
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.12-azure-upi-f28-upgrade-rollback-oldest-supported

Code looks good, let me see some rehearsals of the current revision

**Comment by shellyyang1989** [comment]
Thank you Petr for review! I've run all of the rehearsals. But all failed. Looks like 4.12 rollback is blocked by https://issues.redhat.com/browse/OCPBUGS-25382.

File: ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml [hunk 1]
```diff
@@ -52060,6 +52060,186 @@ periodics:
...
+    containers:
+    - args:
+      - --gcs-upload-secret=/secrets/gcs/service-account.json
+      - --image-import-pull-secret=/etc/pull-secret/.dockerconfigjson
+      - --lease-server-credentials-file=/etc/boskos/credentials
+      - --report-credentials-file=/etc/report/credentials
+      - --secret-dir=/secrets/ci-pull-credentials
+      - --secret-dir=/usr/local/aws-ipi-byo-route53-upgrade-rollback-oldest-supported-cluster-profile
+      - --target=aws-ipi-byo-route53-upgrade-rollback-oldest-supported
+      - --variant=nightly-4.12
```
Comment (LalatenduMohanty) [inline ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml:52100]: @shellyyang1989 we do not need to test the z strema rollback for the nightly builds.

**Review by LalatenduMohanty — CHANGES_REQUESTED** [review]

File: ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml (hunk 1 above)
Comment (LalatenduMohanty) [reply ci-operator/jobs/openshift/release/openshift-release-master-periodics.yaml:52100]: Please check my latest comment in the Jira https://issues.redhat.com/browse/OTA-1084

**Pull request closed without merging**